### PR TITLE
redfish: Split out the SMBIOS table parsing

### PR DESCRIPTION
**Pull request by hughsie**
This makes it more robust against corrupted data, and also adds support
for fuzzing.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/redfish/fu-redfish-common.h [hunk 1]
```diff
@@ -9,14 +9,16 @@
 #include <gio/gio.h>
 
 /* SMBIOS */
-#define REDFISH_SMBIOS_TABLE_TYPE			0x42
+#define REDFISH_SMBIOS_TABLE_TYPE			0x2a	/* 42 */
```
Comment (superm1) [inline plugins/redfish/fu-redfish-common.h:12]: this a mistake?  should split to separate commit i'd say

File: plugins/redfish/fu-redfish-smbios.c [hunk 2]
```diff
@@ -0,0 +1,461 @@
...
+	guint8		 host_ip_address[16];
+	guint8		 host_ip_mask[16];
+	guint8		 service_ip_assignment_type;
+	guint8		 service_ip_address_format;
+	guint8		 service_ip_address[16];
+	guint8		 service_ip_mask[16];
+	guint16		 service_ip_port;
+	guint32		 service_ip_vlan_id;
+	guint8		 service_hostname_len;
+	//gchar		 service_hostname[];
```
Comment (superm1) [inline plugins/redfish/fu-redfish-smbios.c:39]: why commented out?

File: plugins/redfish/fu-redfish-common.h (hunk 1 above)
Comment (hughsie) [reply plugins/redfish/fu-redfish-common.h:12]: Deliberate, unfortunately. a "type 42 table" is 42 decimal. I'll split it out and also backport that line.

File: plugins/redfish/fu-redfish-smbios.c [hunk 3]
```diff
@@ -0,0 +1,461 @@
...
+		break;
+	case REDFISH_INTERFACE_TYPE_PCI_NETWORK:
+		if (!fu_common_read_uint16_safe (buf, bufsz, offset + 0x00,
+						 &self->vid, G_LITTLE_ENDIAN, error))
+			return FALSE;
+		if (!fu_common_read_uint16_safe (buf, bufsz, offset + 0x02,
+						 &self->pid, G_LITTLE_ENDIAN, error))
+			return FALSE;
+		g_debug ("PCI Network Interface: 0x%04x:0x%04x", self->vid, self->pid);
+		break;
```
Comment (superm1) [inline plugins/redfish/fu-redfish-smbios.c:166]: this seems pretty much the same, can you simplify to one case and have g_debug ("%s Network interface:...", `interface_type == foo ? bar : baz`?

Comment (hughsie) [reply plugins/redfish/fu-redfish-smbios.c:166]: The offsets are quite different in v1 and v2 unfortunately :(

Comment (superm1) [reply plugins/redfish/fu-redfish-smbios.c:166]: right but PCI and USB in v1 look the same.

File: plugins/redfish/fu-redfish-smbios.c (hunk 2 above)
Comment (hughsie) [reply plugins/redfish/fu-redfish-smbios.c:39]: Fixed to be less confusing.

File: plugins/redfish/fu-redfish-smbios.c (hunk 3 above)
Comment (hughsie) [reply plugins/redfish/fu-redfish-smbios.c:166]: Done.
